sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -166,20 +168,28 @@ private[sql] object Dataset {
 class Dataset[T] private[sql](
     @transient val sparkSession: SparkSession,
     @DeveloperApi @InterfaceStability.Unstable @transient val queryExecution: QueryExecution,
-    encoder: Encoder[T])
+    encoder: Encoder[T],
+    val sqlText: String = "")
```

Comment:
what's the exact rule you defined to decide whether or not we should propagate the sql text?

Reply:
And how does the SQL shell execute commands? like `SELECT * FROM ...`, does it display all the rows or add a LIMIT before displaying? Generally we should not propagate sql text, as a new DataFrame usually means the plan is changed, the SQL text is not accurate anymore.

Reply:
Thanks for your review. I agree this comment. Before the discuss, let me reproduce the scenario our company met. Team A developed a framework to submit application with sql sentences in a file
> spark-submit --master yarn-cluster --class com.ebay.SQLFramework -s biz.sql

In the biz.sql, there are many sql sentences like
> create or replace temporary view view_a select xx from table ${old_db}.table_a where dt=${check_date};
> insert overwrite table ${new_db}.table_a select xx from view_a join ${new_db}.table_b;
> ...

There is no case like 
`val df = spark.sql("xxxxx")`
`spark.range(10).collect()`
`df.filter(..).count() `

Team B (Platform) need to capture the really sql sentences which are executed in whole cluster, as the sql files from Team A contains many variables. A better way is recording the really sql sentence in EventLog.

Ok, back to the discussion. The original purpose is to display the sql sentence which user inputs. `spark.range(10).collect()` isn't a sql sentence user inputs, either `df.filter(..).count() `. Only "xxxxx" is. So I have two proposals.
1. Change the display behavior, only displays the sql which can trigger action. like "create table", "insert overwrite", etc. Do not care about the select sentence. That won't propagate sql text any more. The test case above won't show anything in SQL ui.
2. Add a SQLCommandEvent and post an event with sql sentence in method SparkSession.sql(), then in the EventLoggingListener, just logging this to eventlog.
3. Open another ticket to add a command option `--sqlfile biz.sql` in spark-submit command. biz.sql must be a file consist by sql sentence. Base this implementation, not only client mode but also cluster mode can use pure sql.

How do you think? @cloud-fan

Reply:
> spark-submit --master yarn-cluster --class com.ebay.SQLFramework -s biz.sql

How does `com.ebay.SQLFramework` process the sql file? just call `spark.sql(xxxx).show` or other stuff?

Reply:
Your speculation is almost right. First call val df = spark.sql(), then separates the sql text with pattern matching to there type: count, limit and other. if count, then invoke the df.showString(2,20). if limit, just invoke df.limit(1).foreach, the last type other will do noting.

Reply:
I have decoupled the sqlText with sql execution. In current implementation, when user invoke spark.sql(xx), it will create a new SparkListenerSQLTextCaptured event to listenerbus. Then in SQLAppStatusListener, the information will be stored and all the sql sentences will display in AllExecutionPage in order with submission time, instead of in each ExecutionPage. I will upload the commit after testing.